### PR TITLE
Detail CONTRIBUTING.md regarding squashing

### PR DESCRIPTION
**Pull request by icecrime**
Cf. https://github.com/docker/docker/pull/11114#issuecomment-81750143

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -217,7 +217,15 @@ mixed into the PR.
 feature branch to update your pull request rather than `merge master`.
 
 Before you make a pull request, squash your commits into logical units of work
-using `git rebase -i` and `git push -f`. After every commit, [make sure the test
+using `git rebase -i` and `git push -f`. A logical unit of work is a consistent
+set of patches that should be reviewed together: for example, upgrading the
+version of a vendored dependency and taking advantage of its now available new
+feature constitute two separate units of work. Implementing a new function and
+calling it in another file constitute a single logical unit of work. The very
+high majory of submissions should have a single commit, so in doubt: squash
```
Comment (estesp) [inline CONTRIBUTING.md]: "so *if* in doubt"?

**Comment by estesp** [comment]
Thank you! I hope this helps..at least it is a good reference.  I know you didn't create the term, but I notice that we are using a capital "L" in "Logical", but I'm not sure why, other than we want it to be seen as a "thing".  Maybe use *italics* where we use it instead, unless we really think capital letter "L" is the right way to denote that?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -217,7 +217,15 @@ mixed into the PR.
...
+down to one.
+
+After every commit, [make sure the test
 suite passes]((https://docs.docker.com/project/test-and-docs/)). Include
```
Comment (tiborvass) [inline CONTRIBUTING.md]: indentation needs to be fixed

**Comment by tiborvass** [comment]
I like this

**Comment by rhvgoyal** [comment]
It might also help to explain that why do we prefer squashed patches instead of broken down ones.

**Comment by icecrime** [comment]
Thanks, updated!

@rhvgoyal I tried to convey this in "consistent set of patches that should be reviewed together". For example, adding a function is generally only half of your intention.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**